### ISSUE TITLE
Maximum IPC message size exceeded on startup

### ISSUE THREAD

**Issue by HarelM** [IsraelHikingMap/Site#1117]
## Bug

![Screenshot_20200107-223930_Israel Hiking Map](https://user-images.githubusercontent.com/3269297/71930106-a7a10780-31a3-11ea-89c1-8c40a644850a.jpg)

## What I expect to happen
This error should not appear

## What really happened
This error was reported by a few users now when the application starts

## What I did that caused the issue - step by step
Open the app - not sure what causes this

## My environment
* Operating System and version: Android
* Web Browser and version:

## Thing I think the developers should know, images, links etc.
Looks like this error message happens when trying to pull out from the database an object that is too large.
The only object stored in the database is the state object.
The only way I can think of making it large is by editing a lot of routes, using a lot of actions to create a large undo stack, see recording etc...
@valleyofdawn @zstadler please let me know if you can reproduce this error, I couldn't unfortunately - It can happen only when opening the app, so all the above steps I mentioned should be done then close the app and then open again, I think...

**Comment by zstadler** [comment]
IMO, the undo stack need not be uploaded to the site.
As a result, undo operations would be limited to the current launch of the app/web site. This is the norm for most apps and sites.

**Comment by zstadler** [comment]
I recieved the attached error after adding several long recordings as private routes, hiding and showing them, and panning the map around. 
![Screenshot_20200108-070442](https://user-images.githubusercontent.com/1304610/71952068-fe2f3580-31e5-11ea-887d-f55f59028cf7.jpg)

**Comment by zstadler** [comment]
The above error seems to be unrelated to the server, as it happens on app launch even in flight mode.
I had to uninstall and re-install the app in order to recover.

**Comment by HarelM** [comment]
I don't think this is the same error according to the message itself, it might be the same root cause though, no way to know...
The error in the initial post is not related to the server, it is a client only issue and can happen when using flight mode when starting the app.
Having said that, using flight mode can introduce other errors that are not related to this issue as well.

>IMO, the undo stack need not be uploaded to the site.
As a result, undo operations would be limited to the current launch of the app/web site. This is the norm for most apps and sites.

I don't think the undo stack can or should be upload to the server but I don't see how this is related to this issue - I need to understand what's causing this error before I can discuss a solution.

**Comment by zstadler** [comment]
Apparently "the database" is local browser storage, and not on the server, as I assumed...

**Comment by zstadler** [comment]
Let's get back to this issue after #1122 is resolved.

**Comment by HarelM** [comment]
A recent report might indicate this is somehow related to low available storage, not sure how accurate the report was though...

**Comment by HarelM** [comment]
When it starts to happen it will reproduce easily, but if you clean the data or re-install the app it doesn't reproduce easily as far as I understand.
What I'm looking for is a reproducible way to make this bug happen when the first reproduction step is: 
1: Install the app, 
And the final steps are: 
N-1: close the app
N: Open the app --> see error message.

From your screenshot, available space doesn't seem to be related to this issue.

**Comment by EdoGeron** [comment]
Regretfully I  cannot provide such flow but maybe I can provide a "state dump" that might assist you?

**Comment by HarelM** [comment]
@EdoGeron thanks, but I don't know how to create one or how to process it...
I have an idea how to solve this issue by cleaning the data when I get this exception, but I would like to use a more precise solution if possible.
I'll try to think of a way to extract the last data that was used and save it to file maybe, which will hopefully give me more info...

**Comment by HarelM** [comment]
This might help:
https://github.com/dfahlander/Dexie.js/issues/468

**Comment by HarelM** [comment]
I think I found a place in the code which might cause this issue - when app initializes it tries to delete all the logs that are older than 3 days.
I have done it by getting all the logs out of the database and then delete them.
I was able to make chrome desktop crash while doing a similar behavior, but when sending a delete request to the database (without pulling out the records) the crash stopped.
Can be seen here:
https://jsfiddle.net/eak3xu6p/7/
I hope this will reduce the appearance of this issue or even solve it completely.
I'll release a version soon.
